src/main/java/ch/njol/skript/lang/ReturnHandler.java

```diff
@@ -0,0 +1,205 @@
+	 * return handler stack
+	 * <br>
+	 * <b>This method may only be called by a {@link Section}</b>
+	 * @throws SkriptAPIException if this return handler is not a {@link Section}
+	 */
+	@NonExtendable
+	default void loadReturnableSectionCode(SectionNode node) {
+		if (!(this instanceof Section))
+			throw new SkriptAPIException("loadReturnableSectionCode called on a non-section object");
+		ParserInstance parser = ParserInstance.get();
+		ReturnHandlerStack stack = parser.getData(ReturnHandlerStack.class);
+		stack.push(this);
+		Section section = (Section) this;
+		try {
+			section.loadCode(node);
+		} finally {
+			stack.pop();
+		}
+	}
+
+	/**
+	 * Loads the code in the given {@link SectionNode} using the same logic as
+	 * {@link Section#loadCode(SectionNode, String, Class[])} and pushes the section onto the
+	 * return handler stack
+	 * <br>
+	 * <b>This method may only be called by a {@link Section}</b>
+	 * @param node the section node
+	 * @param name the name of the event(s) being used
+	 * @param events the event(s) during the section's execution
+	 * @return a returnable trigger containing the loaded section.
+	 * This should be stored and used to run the section one or more times
+	 * @throws SkriptAPIException if this return handler is not a {@link Section}
+	 */
+	@NonExtendable
+	default ReturnableTrigger<T> loadReturnableSectionCode(SectionNode node, String name, Class<? extends Event>[] events) {
+		if (!(this instanceof Section))
+			throw new SkriptAPIException("loadReturnableSectionCode called on a non-section object");
+		ParserInstance parser = ParserInstance.get();
+		ReturnHandlerStack stack = parser.getData(ReturnHandlerStack.class);
+
+		SkriptEvent skriptEvent = new SectionSkriptEvent(name, (Section) this);
+		String previousName = parser.getCurrentEventName();
+		Class<? extends Event>[] previousEvents = parser.getCurrentEvents();
+		Structure previousStructure = parser.getCurrentStructure();
+		List<TriggerSection> previousSections = parser.getCurrentSections();
+		Kleenean previousDelay = parser.getHasDelayBefore();
+
+		parser.setCurrentEvent(name, events);
+		parser.setCurrentStructure(skriptEvent);
+		parser.setCurrentSections(new ArrayList<>());
+		parser.setHasDelayBefore(Kleenean.FALSE);
+		try {
+			return new ReturnableTrigger<>(
+				this,
+				parser.getCurrentScript(),
+				name,
+				skriptEvent,
+				trigger -> {
+					stack.push(trigger);
+					return ScriptLoader.loadItems(node);
+				}
+			);
+		} finally {
+			stack.pop();
+			parser.setCurrentEvent(previousName, previousEvents);
+			parser.setCurrentStructure(previousStructure);
+			parser.setCurrentSections(previousSections);
+			parser.setHasDelayBefore(previousDelay);
+		}
+	}
+
+	/**
+	 * Loads the code in the given {@link SectionNode} into a {@link ReturnableTrigger}.
+	 * <br>
+	 * This is a general method to load a section node without extra logic
+	 * done to the {@link ParserInstance}
```

Comment:
I think it would be good to highlight that the calling code is expected to manage the ParserInstance as required (which may vary depending on where the code being loaded is located and what state the PI is in)